src/qiskit_qasm3_import/expression.py

```diff
@@ -162,81 +176,59 @@ def visit_BinaryExpression(self, node: ast.BinaryExpression):
             raise_from_node(node, f"unsupported binary operation '{node.op.name}'")
         lhs_value, lhs_type = self.visit(node.lhs)
         rhs_value, rhs_type = self.visit(node.rhs)
+        out_type: Union[None, types.Type] = None
+
+        # First, handle the simple implicit promotion rules for the standard numeric types (i.e. not
+        # angle).
+        numeric_t = (types.Int, types.Uint, types.Float)
```

Comment:
I'm looking for a DRYer way to do this. The following is not clean, but works:
```python
numeric_t = _NumericT.__args__
```
A cleaner choice has this at the top:
```python
_IntegerT = Union[types.Never, types.Int, types.Uint]
_NUMERIC_TYPES = (types.Uint, types.Int, types.Float)
_NumericT = Union[*_NUMERIC_TYPES]
```
And then, substitute `_NUMERIC_TYPES` for `numeric_t` below.

Reply:
I'm not sure it's worth worrying about right now - the `Union` type is directly supported as an operand to `isinstance` from like Python 3.9+ or something, and I don't think the tuple unpacking you've got here necessarily works with Mypy - it just does static analysis, and it probably won't recognise that as a type alias any more.